### ISSUE TITLE
[Bug] Sublime Text Key Map syntax bug

### ISSUE THREAD

**Issue by Kristinita** [SublimeText/PackageDev#77]
### Summary

Bug in Sublime Text Key Map syntax. Syntax not working if in my keymap file are arrays:

``` json
    {
            "keys": ["№"],
            "command": "insert_snippet",
            "args": { "contents": "#${0: ${SELECTION/(^ | $)//g} }#" },
            "context": [
                { "key": "selection_empty", "operator": "equal", "operand": false, "match_all": true },
                { "key": "selector", "operator": "equal", "operand": "text.html.markdown", "match_all": true },
                { "key": "preceding_text", "operator": "not_regex_contains", "operand": "#{6}", "match_all": true },
                { "key": "setting.mde.match_header_hashes", "operator": "equal", "operand": true },
                { "key": "selector", "operator": "not_equal", "operand": "markup.raw", "match_all": true }
            ]
        }, {
            "keys": ["№"],
            "command": "insert_snippet",
            "args": { "contents": "#${0: ${SELECTION/(^ | $)//g}}" },
            "context": [
                { "key": "selection_empty", "operator": "equal", "operand": false, "match_all": true },
                { "key": "selector", "operator": "equal", "operand": "text.html.markdown", "match_all": true },
                { "key": "preceding_text", "operator": "not_regex_contains", "operand": "#{6}", "match_all": true },
                { "key": "setting.mde.match_header_hashes", "operator": "equal", "operand": false },
                { "key": "selector", "operator": "not_equal", "operand": "markup.raw", "match_all": true }
            ]
        },
```

If I remove this arrays, syntax works good.
### Expected behavior

![Expected](http://i.imgur.com/RjipyBf.png)
### Actual behavior

![Actual](http://i.imgur.com/zwG9Nma.png)
### Steps to reproduce

`Preferences` → `Key Bindings` → I open my `User` file `sublime-keymap`.
### Environment

**Operating system and version:**
Windows 10.0.14393
**Sublime Text:**
Build 3126

   Thanks.

**Comment by evandrocoan** [comment]
This to be the same problem as mime:

### [Sublime Text Key Map] Breaking for regex expression

Create the file `Default.sublime-keymap` and set the syntax to `Sublime Text Key Map` using this contents and open it with Sublime Text:
```javascript
[
  // extend a docblock by adding an asterisk at the start
  { "keys": ["enter"], "command": "insert_snippet", "args": {"contents": "\n${TM_CURRENT_LINE/^\\s*(\\*\\s*).*$/$1/}"},
    "context": [
      { "key": "setting.auto_indent",   "operator": "equal",          "operand": true,              "match_all": true },
      { "key": "selector",              "operator": "equal",          "operand": "comment.block",   "match_all": true },
      { "key": "auto_complete_visible", "operator": "equal",          "operand": false,             "match_all": true },
      { "key": "preceding_text",        "operator": "regex_contains", "operand": "^\\s*\\*\\s*\\S", "match_all": true }
    ]
  },
  // extend a docblock by adding an asterisk at the start
]
```

### Current Behavior:

The file is correctly parsed by Sublime Text and the `Key Map` is working properly, however the syntax highlighting on the Sublime Text editor is broken for all the remaining file contents after the specified block on the image below.

![image](https://cloud.githubusercontent.com/assets/5332158/20674227/3ff3d044-b56f-11e6-8aa1-7850b02f7841.png)


Originally posted on:

1. https://github.com/sublimehq/Packages/issues/711

**Comment by FichteFoll** [comment]
Fixed in #90.
